src/main/java/dev/openfeature/sdk/ImmutableStructure.java

```diff
@@ -79,9 +79,10 @@ public Map<String, Object> asObjectMap() {
         return attributes
                 .entrySet()
                 .stream()
-                .collect(Collectors.toMap(
-                        Map.Entry::getKey,
-                        e -> convertValue(getValue(e.getKey()))
-                ));
+                // custom collector, workaround for Collectors.toMap in JDK8
+                // https://bugs.openjdk.org/browse/JDK-8148463
+                .collect(HashMap::new,
+                        (accumulated, entry) -> accumulated.put(entry.getKey(), convertValue(entry.getValue())),
+                        HashMap::putAll);
```

Comment:
`Collectors.toMap` doesn't handle null in Java 8! (see bug)

Reply:
wouldn't the `asMap()` and the constructor also suffer from the same problem? 🤔

Reply:
@thisthat @Kavindu-Dodan I found _lots_ of instances of this. I've fixed them all. Fortunately we have good test coverage so even when I made some mistakes with the immutability (forgot to `.clone()`), the tests caught it.